api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -53,13 +54,16 @@ public class StorageController : ControllerBase
     /// <param name="apiOptions"></param>
     /// <param name="fileReferenceService"></param>
     /// <param name="logger"></param>
-    public StorageController(IConnectionHelper connection, IOptions<StorageOptions> storageOptions, IOptions<ApiOptions> apiOptions, IFileReferenceService fileReferenceService, ILogger<StorageController> logger)
+    /// <param name="s3Options"></param>
+    public StorageController(IConnectionHelper connection, IOptions<StorageOptions> storageOptions, IOptions<ApiOptions> apiOptions,
+        IFileReferenceService fileReferenceService, ILogger<StorageController> logger, IOptions<S3Options> s3Options)
     {
         _connection = connection;
         _storageOptions = storageOptions.Value;
         _apiOptions = apiOptions.Value;
         _fileReferenceService = fileReferenceService;
         _logger = logger;
+        _s3Options = s3Options.Value;
```

Comment:
Here you have chosen to use the scoped version of the configuration options.  Which is the standard and in my opinion the correct way to do this.